### PR TITLE
core: Migrate away from config package

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Read a post and actually thought it's time: https://codewithhugo.com/node-configuration-management/

- patch: Migrate away from config package
- Remove config NPM package from package.json

Only did these changes for core. There is config in worker and client as well.

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by vipulgupta2048** [comment]
Controversial ✔️ 
Necessary ✔️ 

What do you folks think?

File: suites/e2e/suite.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const util = require('util');
 // copied from the SV
 // https://github.com/balena-os/balena-supervisor/blob/master/src/config/backends/config-txt.ts
 // TODO: retrieve this from the SDK (requires v16.2.0) or future versions of device contracts
-// https://www.balena.io/docs/reference/sdk/node-sdk/#balena.models.config.getConfigVarSchema
+// https://www.balena.io/docs/reference/sdk/node-sdk/#balena.models.config.nfigVarSchema
```
Comment (klutchell) [inline suites/e2e/suite.js:21]: This seems like a typo

Comment (vipulgupta2048) [reply suites/e2e/suite.js:21]: Seems like it

File: core/lib/components/balena/sdk.js [hunk 2]
```diff
@@ -175,7 +175,7 @@ module.exports = class BalenaSDK {
 		const repo = await this.balena.models.application
 			.get(application)
 			.get('slug');
-		const balenaConfig = await this.balena.models.config.getAll();
+		const balenaConfig = await this.balena.models.config.l();
```
Comment (klutchell) [inline core/lib/components/balena/sdk.js:178]: What does `.l()` do, or is this a typo?

**Review by klutchell — CHANGES_REQUESTED** [review]

File: core/lib/components/balena/sdk.js (hunk 2 above)
Comment (vipulgupta2048) [reply core/lib/components/balena/sdk.js:178]: It seems my bulk changes have had this issue creep in. Thanks for the eagle eye reviewing

File: suites/e2e/suite.js (hunk 1 above)
Comment (klutchell) [reply suites/e2e/suite.js:21]: You had fixed this but now it's wrong again?

**Review by klutchell — CHANGES_REQUESTED** [review]

**Comment by vipulgupta2048** [comment]
My bad, should be good now @klutchell

**Review by klutchell — APPROVED** [review]

**Pull request merged**